assertion update needed on CLI actiovation key tests

most of the assertions are okay, they just try to strictly match the lower case.

```
AssertionError: "'--auto-attach': value must be one of" not found in "[ERROR 2018-05-12T16:25:43 Exception] Error: Option '--auto-attach': Value must be one of true/false, yes/no, 1/0..\n\nSee: 'hammer activation-key update --help'.\nCould not update the activation key:\n  Error: Option '--auto-attach': Value must be one of true/false, yes/no, 1/0..
```

AssertionError: "Error: option '--max-hosts': numeric value is required" not found in 'Failed to create ActivationKey with data:\n{\n  "content-view": null,\n  "content-view-id": null,\n  "description": null,\n  "lifecycle-environment": null,\n  "lifecycle-environment-id": null,\n  "max-hosts": "",\n  "name": "2ICmDcBBmc",\n  "organization": null,\n  "organization-id": "334",\n  "organization-label": null,\n  "unlimited-hosts": null\n}\nCommand "activation-key create" finished with return_code 65\nstderr contains following message:\n[ERROR 2018-05-18T15:28:04 API] 422 Unprocessable Entity\n[ERROR 2018-05-18T15:28:04 Exception] Validation failed: Max hosts cannot be nil\nCould not create the activation key:\n  Validation failed: Max hosts cannot be nil\n[ERROR 2018-05-18T15:28:04 Exception] \n\nRestClient::UnprocessableEntity (422 Unprocessable Entity)

there's another problem with the assertions now on passing `--name="\t"`:
```
AssertionError: '\t' not found in 'Failed to create ActivationKey with data...
```

the reason is, that our test treats "\t" as a string with length >0 thus expects different validation-failed message.

--- a/tests/foreman/cli/test_activationkey.py
+++ b/tests/foreman/cli/test_activationkey.py
@@ -288,19 +288,14 @@ class ActivationKeyTestCase(CLITestCase):
             with self.subTest(name), self.assertRaises(
                     CLIFactoryError) as raise_ctx:
                 self._make_activation_key({u'name': name})
-            content_to_validate = [u'Validation failed:']
-            if len(name) == 0:
-                content_to_validate.append(
+            if name in ['', ' ', '\t']:
+                self.assert_error_msg(
+                    raise_ctx,
                     u'Name must contain at least 1 character')
-            elif len(name) > 255:
-                content_to_validate.append(
+            if len(name) > 255:
+                self.assert_error_msg(
+                    raise_ctx,
                     u'Name is too long (maximum is 255 characters)')
-            else:
-                content_to_validate.append(name)
-            self.assert_error_msg(
-                raise_ctx,
-                *content_to_validate
-            )
 
     @tier1
     def test_negative_create_with_usage_limit_with_not_integers(self):
@@ -320,10 +315,24 @@ class ActivationKeyTestCase(CLITestCase):
             if not value.isdigit()
         ]
         invalid_values.append(0.5)
-        self.assert_negative_create_with_usage_limit(
-            invalid_values,
-            u"Error: option '--max-hosts': numeric value is required"
-        )
+        for limit in invalid_values:
+            with self.subTest(limit), self.assertRaises(
+                    CLIFactoryError) as raise_ctx:
+                self._make_activation_key({u'max-hosts': limit})
+            if type(limit) is int:
+                if limit < 1:
+                    self.assert_error_msg(
+                        raise_ctx,
+                        u'Max hosts cannot be less than one')
+            if type(limit) is str:
+                if limit in ['', ' ', '\t']:
+                    self.assert_error_msg(
+                        raise_ctx,
+                        u'Max hosts cannot be nil')
+                if len(limit) > 1 and limit.isalpha():
+                    self.assert_error_msg(
+                        raise_ctx,
+                        u'Max hosts is not a number')
 
     @tier1
     def test_negative_create_with_usage_limit_with_invalid_integers(self):

